airbyte-cdk/python/airbyte_cdk/sources/file_based/file_based_source.py

```diff
@@ -151,8 +145,8 @@ def streams(self, config: Mapping[str, Any]) -> List[Stream]:
         else:
             # During `check` operations we don't have a catalog so cannot create a state manager.
             # Since the state manager is only required for incremental syncs, this is fine.
+            # (for now - it will be an issue upstack when everything is incremental, but we'll deal with that then)
```

Comment:
These none values when we run a check and don't have catalogs or state are so painful. But it also does point more and more that we're moving in the right direction that things like `sync_mode` shouldn't even be a consideration by the connector methods